katpoint/projection.py

```diff
@@ -127,26 +131,234 @@
 """
 from __future__ import print_function, division, absolute_import
 
+import threading
+import contextlib
+
 import numpy as np
 
+# --------------------------------------------------------------------------------------------------
+# --- Handling out-of-range inputs
+# --------------------------------------------------------------------------------------------------
+
+
+class OutOfRangeError(ValueError):
+    """A numeric value is out of range."""
+
+
+_out_of_range_cvar = threading.local()
+_out_of_range_cvar.treatment = 'raise'
+
```

Comment:
I'd prefer if the default was nan... but I guess I wont win that battle...

Reply:
Backwards compatibility FTW 😄 

Just put the following at the top of your script:

```
katpoint.projection.set_out_of_range_treatment('nan')
```